File: TelerikUIExamplesInCSharp/Examples/Chart/AxisTypes/DateTimeCategoryAxis.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;


namespace Examples
{
	public class DateTimeCategoryAxis: ExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.ExampleBounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			var xAxis = new TKChartDateTimeCategoryAxis ();
			var formatter = new NSDateFormatter ();
			formatter.DateFormat = "d MMM";
			xAxis.LabelFormatter = formatter;
			xAxis.Style.MajorTickStyle.TicksOffset = -3;
			xAxis.Style.MajorTickStyle.TicksHidden = false;
			xAxis.Style.MajorTickStyle.TicksWidth = 1.5f;
			xAxis.Style.MajorTickStyle.TicksFill = new TKSolidFill (new UIColor (203 / 255.0f, 203 / 255.0f, 203 / 255.0f, 1f));
			xAxis.Style.MajorTickStyle.MaxTickClippingMode = TKChartAxisClippingMode.Visible;
			xAxis.PlotMode = TKChartAxisPlotMode.BetweenTicks;

			var yAxis = new TKChartNumericAxis (new NSNumber(320), new NSNumber(360));
			yAxis.MajorTickInterval = 20;

			var series = new TKChartCandlestickSeries(StockDataPoint.LoadStockPoints(10).ToArray());
			series.GapLength = 0.6f;
			series.XAxis = xAxis;
			chart.YAxis = yAxis;

			chart.AddSeries (series);
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/DateTimeCategoryAxis.swift
//
//  DateTimeCategoryAxis.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

class DateTimeCategoryAxis: ExampleViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let chart = TKChart(frame:self.exampleBoundsWithInset)
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        
        let xAxis = TKChartDateTimeCategoryAxis()
        let formatter = NSDateFormatter()
        formatter.dateFormat = "d MMM"
        xAxis.labelFormatter = formatter
        xAxis.style.majorTickStyle.ticksOffset = -3
        xAxis.style.majorTickStyle.ticksHidden = false
        xAxis.style.majorTickStyle.ticksWidth = 1.5
        xAxis.style.majorTickStyle.ticksFill = TKSolidFill(color:UIColor(red: 203/255.0, green: 203/255.0, blue: 203/255.0, alpha: 1))
        xAxis.style.majorTickStyle.maxTickClippingMode = TKChartAxisClippingMode.Visible
        xAxis.setPlotMode(TKChartAxisPlotMode.BetweenTicks)
        
        let yAxis = TKChartNumericAxis(minimum:320, andMaximum:360)
        yAxis.majorTickInterval = 20
        
        let series = TKChartCandlestickSeries(items:StockDataPoint.stockPoints(10))
        series.gapLength = 0.6
        series.xAxis = xAxis
        chart.yAxis = yAxis
        
        chart.addSeries(series)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
